test-common/src/main/scala/org/scalajs/testcommon/RunMuxRPC.scala

```diff
@@ -0,0 +1,81 @@
+package org.scalajs.testcommon
+
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.util.Try
+
+import java.util.concurrent.ConcurrentHashMap
+
+import FutureUtil.futureFromTry
+
+/** Helper above an [[RPCCore]] that allows to multiplex between runs. */
+private[scalajs] final class RunMuxRPC(rpc: RPCCore) {
+  import RunMux.RunId
+
+  // Helper types
+  type MuxRPCEndpoint[T] = RPCEndpoint { type Req = RunMux[T] }
+  type MuxMsgEndpoint[T] = MsgEndpoint { type Msg = RunMux[T] }
+
+  /** Multiplexer map.
+   *
+   *  Access to the outer map needs to synchronized.
+   *  Access to the inner map only needs to be synchronize for writing.
+   */
+  private[this] val mux = mutable.Map.empty[RPCCore.OpCode, ConcurrentHashMap[RunId, _]]
+
+  def call[Req](ep: MuxRPCEndpoint[Req], runId: RunId)(req: Req): Future[ep.Resp] =
+    rpc.call(ep)(new RunMux(runId, req))
```

Comment:
Why is the signature of this method not
```scala
def call(ep: RPCEndpoint, runId: RunId)(req: ep.Req): Future[ep.Resp]
```
i.e., exploiting the path-dependent type `eq.Req`? Same for the other methods below.